src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,226 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.scale.x = 0.3f;
+    this->dyna.actor.scale.y = 0.1f;
+    this->dyna.actor.scale.z = 0.3f;
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &gStoneTowerTemplePunchablePillarCol);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    if (!BGIKANADHARMA_IS_CHILD(&this->dyna.actor)) {
+        f32 segmentY = this->dyna.actor.world.pos.y;
+        s32 numSegments = BGIKANADHARMA_NUM_SEGMENTS(&this->dyna.actor);
+        s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC5C4.s")
+        for (i = 0; i < numSegments; i++) {
+            segmentY += 60.0f;
+            Actor_SpawnAsChildAndCutscene(&globalCtx2->actorCtx, globalCtx, ACTOR_BG_IKANA_DHARMA,
+                                          this->dyna.actor.world.pos.x, segmentY, this->dyna.actor.world.pos.z,
+                                          this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y,
+                                          this->dyna.actor.shape.rot.z, BGIKANADHARMA_FLAG_IS_CHILD,
+                                          this->dyna.actor.cutscene, this->dyna.actor.unk20, NULL);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC5E0.s")
+        this->dyna.actor.bgCheckFlags |= 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC758.s")
+    BgIkanaDharma_BeginNormalState(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC790.s")
+void BgIkanaDharma_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkanaDharma* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC7EC.s")
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    if (sFirstHitBgIkanaDharma == this) {
+        sFirstHitBgIkanaDharma = NULL;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC808.s")
+void BgIkanaDharma_BeginNormalState(BgIkanaDharma* this) {
+    this->actionFunc = BgIkanaDharma_UpdateNormalState;
+    this->dyna.actor.speedXZ = 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Update.s")
+void BgIkanaDharma_UpdateNormalState(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    s32 wasHit = (this->collider.base.acFlags & AC_HIT) != 0;
+    Player* player = GET_PLAYER(globalCtx);
+    BgIkanaDharma* this2 = this;
+    s32 tempAngle1;
+    s32 tempAngle2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Draw.s")
+    if (wasHit) {
+        this->collider.base.acFlags &= ~AC_HIT;
+    }
+
+    if (wasHit && sFirstHitBgIkanaDharma == NULL) {
```

Comment:
Weird, almost looks like `wasHit` was supposed to be in the condition directly, but its used twice, dont know if IDO can optimized calling that once and saving the result

Reply:
Yeah I tried to remove it and couldn't. There might be a way to make it work but since I just picked this PR up to get this actor in, I didn't try very hard.